Add gnuradio 3.10 - 3.11 support

bkerler opened the pull request:
Tested with gnuradio 3.11. Adds additional patches from other pull requests.
I had to restructure the directories in order to match the official gnuradio module structure.

bkerler committed:
Merge pull request #1 from ptrkrysik/master

Update to last commit

bkerler committed:
Merge pull request #2 from ptrkrysik/master

Update

bkerler committed:
Support gnuradio 3.10/3.11

bkerler committed:
Add patches from @Deligrim

bkerler committed:
Fix path issues

bkerler committed:
Additional fixes

bkerler committed:
More fixes

bkerler committed:
Fixes compilation issues

velichkov commented on python/gsm/receiver/multiarfcns_receiver.py:
```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python2
```
Either change this to python3 or remove the whole shebang as this is not really an executable.

bkerler replied:
Solved.

velichkov commented on apps/grgsm_livemon.py:
```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env python3
```
Remove `grgsm_livemon.py` from the repository as it is generated/compiled from `grgsm_livemon.grc` during `build`

bkerler replied:
Solved.

velichkov commented on apps/grgsm_livemon_headless.py:
```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python3
```
Remove `grgsm_livemon_headless.py` from the repository as it is generated/compiled from `grgsm_livemon_headless.grc` during build.

bkerler replied:
Solved.

velichkov reviewed (changes requested):
Hi @bkerler,

Welcome to `gr-gsm` and thanks for taking the time to port it to the latest gnuradio version. Your effort would be much appreciated.

> Tested with gnuradio 3.11. 

Have you tested it with 3.10 as you branch is called `maint-3.10` and not `maint-3.11`. Are you familiar with `docker` and can you provide some Dockerfiles similar to `tests/dockerfiles/Ubuntu_20_04.docker` and `tests/dockerfiles/Debian_testing.docker`. Previously we have used TravisCI for testing but right now there is something wrong with it but sill we can use these the Dockerfiles for local testing.

I still haven't finished the whole review as there are quite a lot of changes but here are some comments

bkerler committed:
Delete grgsm_livemon.py

bkerler committed:
Delete grgsm_livemon_headless.py

bkerler committed:
Update multiarfcns_receiver.py

bkerler commented:
Hi, thanks for pointing them out, I've resolved the issues as you proposed.

bkerler commented:
I've tested the module with 3.10 and 3.11. It should work with 3.9 as well.

bkerler commented:
Regarding docker : I'm not a big fan of it. But I will be able to provide a local installation script/package soon (which doesn't interfer with the installed system).

bkerler committed:
Update firdes to window

bkerler committed:
Fix grc

AkechiShiro commented:
Sorry to ask this question, I'm aware that on open source there is strictly no ETA, but when is this roughly expected to land in master, in a week or two perhaps ?

bkerler commented:
Seems the code is incompatible with 3.9 as gnuradio added incompatible cmake options on 3.10.

bkerler commented:
For those interested into support for 3.9, I did a backport here: https://github.com/bkerler/gr-gsm/tree/maint-3.9

bkerler committed:
Fix ffft filter

axilirator commented:
First of all, thanks for your effort! I tried to migrate to 3.9 myself, but quickly gave up.

This Pull Request mixes up several things. The topic is "Add gnuradio 3.10 - 3.11 support", however you're also adding a new block (Multi Arfcns GSM Receiver) here. Ideally this should be done in a separate Pull Request. Also, small commits like https://github.com/ptrkrysik/gr-gsm/pull/600/commits/dc7774cf99daebb7514ca0d76abda9b0597679dc, https://github.com/ptrkrysik/gr-gsm/pull/600/commits/5accd4a19bb1c8e766dafc860811943f6b5c2230, https://github.com/ptrkrysik/gr-gsm/pull/600/commits/14c94dc9d76b8813973a347caedc28bbd618802b, and https://github.com/ptrkrysik/gr-gsm/pull/600/commits/d6670963580f575d3cdb4236353a51cfd56caf7b should be squashed into the main commit because IMO it makes a little sense to have them in the git log.

bkerler commented:
Yes, I've added other existing pull requests where it seemed reasonable to integrate them. It wasn't planned to write a pr in the beginning, that's why I didn't keep it separately, sorry. I could separate it from the port but that might take a while if that's really required.

bkerler commented:
The "path" issue is required to match the gnuradio 3.9 project structure and to avoid conflicts with existing non-gnuradio python packets. Backporting to <3.9 doesn't make sense to me to be honest.

bkerler committed:
Revert dda008c998c9655bb55c2836c19a233d55e82865

bkerler commented:
I've removed the multi arfcns stuff.

bkerler committed:
Update grgsm_scanner

Remove gsm import

bkerler committed:
Update grgsm_trx

bkerler committed:
Update grgsm_livemon.grc

alphafox02 commented:
Tested this multiple different ways and found that both the maint-3.10 and maint-3.10_with_multiarfcn generated results when tested with the catcher script on a 20.04 aarch64 build with GR 3.10.1.1 from PPA. This was also using gr-osmosdr from source.

Awesome job @bkerler 

Only thing I'm noticing is there's no output to terminal when decoding is happening.

bkerler committed:
Fix device detection

bkerler committed:
Fix grc message print issue

AkechiShiro commented:
Hey @bkerler is this branch buildable, can I test it ? I need to make the latest Gnuradio work with gr-gsm

alphafox02 commented:
It’s buildable and works w/ GR 3.10. I’ve been using it, I think the only thing that maybe has a problem is the decode tool. But livemon, scanner etc works.

AkechiShiro commented:
@alphafox02 just built it and installed it according to the wiki intrusctions, and I'm having an error, not sure how am I supposed to use it ? I have to use it directly inside gnuradio ?
```
> grgsm_scanner
Traceback (most recent call last):
  File "/usr/local/bin/grgsm_scanner", line 33, in <module>
    from gnuradio import gsm
ImportError: cannot import name 'gsm' from 'gnuradio' (/usr/lib/python3.10/site-packages/gnuradio/__init__.py)
```
EDIT: I'm not using Debian, I'm under ArchLinux

alphafox02 commented:
I’ve only built it into an aarch64 20.04 based image and use it as shown here. GR-GSM w/ GNU Radio 3.10 (RTLSDR, Pi4, LimeSDR, OSMO-NITB). I’m not real familiar with arch.
https://youtu.be/uU7JrwAgmuE

AkechiShiro commented:
Thanks nonetheless, will probably do the same then using maybe docker or a chroot but it's pretty annoying...

I don't know when is this supposed to land in Gnuradio, can we help for it to land in Gnuradio faster ? @bkerler ?

bkerler commented:
Not sure if you're using the right branch of my repo,  the multi arfcn branch should build and work fine.

AkechiShiro commented:
@bkerler I'm using the `maint-3.10`

bkerler commented:
Maint-3.10 doesn't have the needed fixes right now as the main author wanted me to remove the additional patches

AkechiShiro commented:
So @bkerler it is the other one branch that I have to use thanks then.

velichkov commented on apps/CMakeLists.txt:
```diff
@@ -25,29 +25,29 @@ add_subdirectory(helpers)
-OPTION(ENABLE_GRGSM_LIVEMON "Compile grgsm_livemon" ON)
-OPTION(ENABLE_GRGSM_LIVEMON_HEADLESS "Compile grgsm_livemon_headless" ON)
+set(gsm_flowgraphs "")
+#OPTION(ENABLE_GRCC "Compile the flowgraphs with grcc" ON)
```
Hi @bkerler,

Why did you comment out `ENABLE_GRCC` option? This seems to break generation of `grgsm_livemon` and `grgsm_livemon_headless` as reported in the gr-gsm's mailing list.

https://groups.google.com/g/gr-gsm/c/F1zUJz2DVEc/m/w3m85luRIQAJ

bkerler replied:
I uncommented it because otherwise the package fails to compile due to gnuradio 3.9 package compiling changes. So far I couldn't figure out how to fix it as it refers to packages to import that can't be imported because they aren't compiled.

velichkov replied:
Can you post the exact error messages and list the packages that aren't compiled?

alphafox02 commented:
I just manually generated both after install and placed them in the /bin dir. Didn’t solve the initial problem, but that’s how I’ve been using it on my pi with 3.10. Livemon and scanner work great. Only issue I still recall having is with grgsm_decode and some error, but I’ve forgotten what it was.

pixkike commented:
IT did not work or I miss some dependency ? I install fnuradio 3.10 and install grgsm from  @bkerler  source 
Buld and installation done But it didn't work Please Someone tell me how do I fix this 

`
grgsm_decode                                                                                               1 ⨯
Traceback (most recent call last):
  File "/usr/local/bin/grgsm_decode", line 31, in <module>
    from gnuradio import gsm
  File "/usr/local/lib/python3.10/dist-packages/gnuradio/gsm/__init__.py", line 46, in <module>
    from .gsm_python import *
ImportError: libgnuradio-gsm.so.1.0.0git: cannot open shared object file: No such file or directory
`

BTW  I can't run "grgsm_livemon" it show me " command  not found "
I use kali 2022

alphafox02 commented:
@bkerler the person above mentioned a grgsm_decode issue, which if I recall is the same issue I had. @pixkike find the flow graph in that source you pulled down, open it in gnuradio and generate the Python file. Make it executable, remove the .py and put it in bin. That’s from memory but I think it’s about what I did.

bkerler commented:
Yes that's how it works right now, but that's not the way it's intended to work

velichkov committed:
Fix building of grgsm_livemon and grgsm_livemon_headless

velichkov committed:
Update the Ubuntu container to 22.04

velichkov committed:
In unit tests import gsm module from gnuradio

velichkov committed:
Add Fedora 36 container

bkerler commented:
here's the error : 
```
**************************************************
3 errors from flowgraph:
Param - Cell allocation(cell_allocation):
	Value "[arfcn.downlink2arfcn(fc)]" cannot be evaluated:
	name 'arfcn' is not definedParam - Cell allocation(cell_allocation):
	Expression None is invalid for type'float_vector'.Param - Import(imports):
	Import "import gnuradio.gsm.arfcn as arfcn" failed.
**************************************************
Compilation error
```

bkerler commented:
it's basically commit 6eea94f7de77dac2e547cb629e61a82ec0a35bc4 that fails.

velichkov commented:
> 	Import "import gnuradio.gsm.arfcn as arfcn" failed.

Thanks. I've fixed this and opened a PR in your fork https://github.com/bkerler/gr-gsm/pull/3. Now I'm trying to fix the failure in `qa_burst_file_source` test.

bkerler committed:
Merge pull request #3 from velichkov/maint-3.10-fixes

Fix building of grgsm_livemon and grgsm_livemon_headless

petterreinholdtsen committed:
Make the build reproducible

Patch 1020-reproducible-build.patch from Debian package, written by
Chris Lamb to solve https://bugs.debian.org/891405 .

bkerler committed:
Merge pull request #4 from petterreinholdtsen/reproducable-build

Make the build reproducible

axistwelve commented:
Hi,

I am still experiencing issues on livemon compilation (I am using Kali 2022) and maint-3.10-fixes and Gnun 3.10.4
ii  gnuradio                               3.10.4.0-2+b1                          armhf        GNU Radio Software Radio Toolkit
ii  gnuradio-dev                           3.10.4.0-2+b1                          armhf        GNU Software Defined Radio toolkit development
`cd /usr/local/src/IMSI-catcher/gr-gsm/build/apps && /usr/bin/cmake -E env PYTHONPATH="/usr/local/src/IMSI-catcher/gr-gsm/build/test_modules" GRC_BLOCKS_PATH=/usr/local/src/IMSI-catcher/gr-gsm/grc CMAKE_BINARY_DIR=/usr/local/src/IMSI-catcher/gr-gsm/build /usr/bin/grcc -o /usr/local/src/IMSI-catcher/gr-gsm/build/apps /usr/local/src/IMSI-catcher/gr-gsm/apps/grgsm_livemon_headless.grc
>>> Warning: vocoder_codec2_decode_ps - option_attributes are for enums only, ignoring
>>> Warning: vocoder_codec2_encode_sp - option_attributes are for enums only, ignoring
<<< Welcome to GNU Radio Companion Compiler 3.10.4.0 >>>

Block paths:
        /usr/local/src/IMSI-catcher/gr-gsm/grc
        /usr/share/gnuradio/grc/blocks

>>> Loading: /usr/local/src/IMSI-catcher/gr-gsm/apps/grgsm_livemon_headless.grc
>>> Load Error: /usr/local/src/IMSI-catcher/gr-gsm/apps/grgsm_livemon_headless.grc: Flowgraph invalid
**************************************************
4 errors from flowgraph:
Param - Cell allocation(cell_allocation):
        Value "[arfcn.downlink2arfcn(fc)]" cannot be evaluated:
        name 'arfcn' is not definedParam - Cell allocation(cell_allocation):
        Expression None is invalid for type'float_vector'.Param - Import(imports):
        Import "import gnuradio.gsm as gsm" failed.Param - Import(imports):
        Import "from gnuradio.gsm import arfcn" failed.
**************************************************
Compilation error
`

axistwelve commented:
> I just manually generated both after install and placed them in the /bin dir. Didn’t solve the initial problem, but that’s how I’ve been using it on my pi with 3.10. Livemon and scanner work great. Only issue I still recall having is with grgsm_decode and some error, but I’ve forgotten what it was.

Hey @alphafox02 so if I comment out the code as per above, how do I manually generate the two items post build (as a works around for the issue). Best regards

alphafox02 commented:
I’m pretty certain @bkerler fork maint_3.10 branch has all this sorted out now and the files are generated after the make install. Not at my computer at the moment but will check this evening.

bkerler committed:
Fix decoding issue with gr3.10

bkerler commented:
Fixed with latest commits on both multiarfcn and main-3.10 branches.

dstuart commented:
Hi, So I am using rasberry Pi
No LSB modules are available.
Distributor ID:	Kali
Description:	Kali GNU/Linux Rolling
Release:	2022.3
Codename:	kali-rolling

Python 3.10.5
ii  gnuradio                               3.10.4.0-2+b1                          armhf        GNU Radio Software Radio Toolkit
ii  gnuradio-dev                           3.10.4.0-2+b1                          armhf        GNU Software Defined Radio toolkit development
ii  gr-osmosdr                             0.2.3-7+b1                             armhf        Gnuradio blocks from the OsmoSDR project

With 
* maint-3.10 (I have also tried   maint-3.10_with_multiarfcn) on  https://github.com/bkerler/gr-gsm.git

It still wont compile livemon. I can make if work, commenting out the CMakeLists thing
OPTION(ENABLE_GRGSM_LIVEMON "Compile grgsm_livemon" ON)
OPTION(ENABLE_GRGSM_LIVEMON_HEADLESS "Compile grgsm_livemon_headless" ON)
set(gsm_flowgraphs "")
#OPTION(ENABLE_GRCC "Compile the flowgraphs with grcc" ON)

The error I get is the same as per above
grcc -o /gr-gsm/build/apps /gr-gsm/apps/grgsm_livemon_headless.grc
>>> Warning: vocoder_codec2_decode_ps - option_attributes are for enums only, ignoring
>>> Warning: vocoder_codec2_encode_sp - option_attributes are for enums only, ignoring
<<< Welcome to GNU Radio Companion Compiler 3.10.4.0 >>>

Block paths:
        /root/.grc_gnuradio
        /usr/share/gnuradio/grc/blocks
        /usr/local/share/gnuradio/grc/blocks

>>> Loading: /gr-gsm/apps/grgsm_livemon_headless.grc
>>> Load Error: /gr-gsm/apps/grgsm_livemon_headless.grc: Flowgraph invalid
**************************************************
4 errors from flowgraph:
Param - Cell allocation(cell_allocation):
        Value "[arfcn.downlink2arfcn(fc)]" cannot be evaluated:
        name 'arfcn' is not definedParam - Cell allocation(cell_allocation):
        Expression None is invalid for type'float_vector'.Param - Import(imports):
        Import "import gnuradio.gsm as gsm" failed.Param - Import(imports):
        Import "from gnuradio.gsm import arfcn" failed.
**************************************************
Compilation error


(not sure of relevant) If I do a simple import of the gnuradio files I get this
>>> import gnuradio.gsm as gsm
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/gnuradio/gsm/__init__.py", line 46, in <module>
    from .gsm_python import *
ImportError: generic_type: type "burst_file_source" referenced unknown base type "gr::block"

>>> from gnuradio.gsm import arfcn
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/gnuradio/gsm/__init__.py", line 46, in <module>
    from .gsm_python import *
ImportError: generic_type: type "burst_file_source" referenced unknown base type "gr::block"

Please help

dstuart commented:
@alphafox02 Did you manage to test this to see if it's working for you now? As per above I commented it out and managed to get to a make install, thinking once the files were in place it might all work for the grc's but it still giving me pain. How do you manually generate grgsm_livemon and grgsm_livemon_headless ?

alphafox02 commented:
If I recall, I’ve not had to do anything manual nowadays. I’ve built the non multi arfcn branch on x86_64 and aarch64 now. If you look up DragonOS Pi64 you’ll find the image prebuilt with 3.10.4 + grgsm etc. maybe take a peak and see what’s different about it.

alphafox02 commented:
To Manually generate you can open the flow graphs with gnuradio companion, click the button to generate the py file. Remove the .py extension, make then executable with chmod, and place then I think in /usr/local/bin, but you should’nt have to do that anymore.

dstuart commented:
Ok great thanks for the info. will definitely look into both of those things

alphafox02 commented:
@dstuart @bkerler fyi - I tried to build the multi arfcn branch for 3.10 and experienced this as well https://github.com/ptrkrysik/gr-gsm/pull/600#issuecomment-1329066124 w/ GR 3.10.4/22.04 

It does not occur when using the non multi arfcn branch w/ GR 3.10.

bkerler committed:
Fix issues with radio_if.py and grgsm_trx

DethCount commented:
Fixed by updating symlink 
lrwxrwxrwx 1 root root      27 mai   18 10:14 libgnuradio-gsm.so.1.0.0git -> libgnuradio-gsm.so.g199fb3f
instead of
lrwxrwxrwx 1 root root      27 mai   18 10:14 libgnuradio-gsm.so.1.0.0git -> libgnuradio-gsm.so.199fb3f

Dieg0x17 commented:
Tras compilar y tratar de importar el modulo gsm, aparece el siguiente fallo

```
>>> import gnuradio.gsm.arfcn
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/dist-packages/gnuradio/gsm/__init__.py", line 46, in <module>
    from .gsm_python import *
ImportError: libgnuradio-gsm.so.1.0.0git: cannot open shared object file: No such file or directory
```
en el  **/usr/local/lib/python3.11/dist-packages/gnuradio/gsm/_\_init__.py** esta sección

```
# import pybind11 generated symbols into the gsm namespace
try:
    # this might fail if the module is python-only
    from .gsm_python import *
except ModuleNotFoundError:
    pass

```

Por algun motivo no encuentra la siguiente libreria al compilar
```      
libgnuradio-gsm.so.1.0.0git => not found
```

Esta otra libreria parece que no esta enlazada con aquella.
```
ldd -r gsm_python.cpython-311-x86_64-linux-gnu.so 
        linux-vdso.so.1 (0x00007fff364fe000)
        libgnuradio-gsm.so.1.0.0git => not found
        libgnuradio-runtime.so.3.10.10 => /lib/x86_64-linux-gnu/libgnuradio-runtime.so.3.10.10 (0x00007f9378e01000)
        libgnuradio-pmt.so.3.10.10 => /lib/x86_64-linux-gnu/libgnuradio-pmt.so.3.10.10 (0x00007f9378da5000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9378b40000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9378b13000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f937892c000)
        libvolk.so.3.1 => /lib/x86_64-linux-gnu/libvolk.so.3.1 (0x00007f937864e000)
        libboost_thread.so.1.83.0 => /lib/x86_64-linux-gnu/libboost_thread.so.1.83.0 (0x00007f937862b000)
        libspdlog.so.1.12 => /lib/x86_64-linux-gnu/libspdlog.so.1.12 (0x00007f93785a0000)
        libboost_program_options.so.1.83.0 => /lib/x86_64-linux-gnu/libboost_program_options.so.1.83.0 (0x00007f9378533000)
        libfmt.so.9 => /lib/x86_64-linux-gnu/libfmt.so.9 (0x00007f9378513000)
        libgmp.so.10 => /lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f9378489000)
        libunwind.so.8 => /lib/x86_64-linux-gnu/libunwind.so.8 (0x00007f937846d000)
        libthrift-0.19.0.so => /lib/x86_64-linux-gnu/libthrift-0.19.0.so (0x00007f93783a9000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f93782c7000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f93790b1000)
        liborc-0.4.so.0 => /lib/x86_64-linux-gnu/liborc-0.4.so.0 (0x00007f937820e000)
        liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f93781dc000)
        libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007f93780e7000)
        libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007f9377b73000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f9377b54000)
        libzstd.so.1 => /lib/x86_64-linux-gnu/libzstd.so.1 (0x00007f9377a93000)
undefined symbol: PyInstanceMethod_Type (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_ValueError      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm23burst_timeslot_splitter4makeEv    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm12message_sink4makeEv       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm16burst_fnr_filter4makeENS0_11filter_modeEj (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _Py_TrueStruct        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_IndexError      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm17burst_type_filter4makeERKSt6vectorIhSaIhEE        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm14bursts_printer4makeESt10shared_ptrIN3pmt8pmt_baseEEbbbb   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm11extract_cmc4makeEv        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm10burst_sink4makeEv (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm15message_printer4makeESt10shared_ptrIN3pmt8pmt_baseEEbbb   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm13txtime_setter4makeEjmdmddd        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm13major_versionB5cxx11Ev    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCapsule_Type        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm34controlled_fractional_resampler_cc4makeEff        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyModule_Type (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm18dummy_burst_filter4makeEv (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm8receiver4makeEiRKSt6vectorIiSaIiEES6_b     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm7versionB5cxx11Ev   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm10msg_to_tag4makeEv (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm17cx_channel_hopper4makeERKSt6vectorIiSaIiEEii      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _Py_NoneStruct        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm24control_channels_decoder4makeEv   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm11api_versionB5cxx11Ev      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_MemoryError     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyType_Type   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyByteArray_Type      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm19extract_system_info4makeEv        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm21controlled_rotator_cc4makeEd      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCFunction_Type      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_OverflowError   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm19preprocess_tx_burst4makeEv        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyProperty_Type       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm19collect_system_info4makeEv        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm15burst_file_sink4makeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm24uplink_downlink_splitter4makeEv   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm11tmsi_dumper4makeEv        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm13tch_f_decoder4makeENS0_8tch_modeEb        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_BufferError     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm17message_file_sink4makeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm20tch_h_chans_demapper4makeEjj      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm10build_dateB5cxx11Ev       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_RuntimeError    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm17fn_time_delta_cppEjSt4pairIydEjS2_jj      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm11gen_test_ab4makeEv        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _Py_NotImplementedStruct      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm28burst_sdcch_subslot_splitter4makeENS0_13splitter_modeE    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyBaseObject_Type     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_TypeError       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm10decryption4makeERKSt6vectorIhSaIhEEj      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyMethod_Type (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm12trx_burst_if4makeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES9_S9_    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm13minor_versionB5cxx11Ev    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _Py_FalseStruct       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_Type   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm29universal_ctrl_chans_demapper4makeEjRKSt6vectorIiSaIiEES6_S6_S6_S6_S6_    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm28extract_immediate_assignment4makeEbbb     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm17burst_file_source4makeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm14message_source4makeERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EE      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm12burst_source4makeERKSt6vectorIiSaIiEES6_RKS2_INSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaISC_EE        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm20tch_f_chans_demapper4makeEj       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyFloat_Type  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm13maint_versionB5cxx11Ev    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_SystemError     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyExc_ImportError     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm16burst_to_fn_time4makeEv   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm22extract_assignment_cmd4makeEv     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm20clock_offset_control4makeEffj     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm13tch_h_decoder4makeEjNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEb (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm26burst_sdcch_subslot_filter4makeENS0_19subslot_filter_modeEj       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm21burst_timeslot_filter4makeEj      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _ZN2gr3gsm19message_file_source4makeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_GenericGetDict       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_GenericSetDict       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyTuple_SetItem       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_Repr (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyLong_AsLong (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyLong_FromSsize_t    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyByteArray_Size      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_Call (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyNumber_And  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_NormalizeException      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyInstanceMethod_New  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyEval_AcquireThread  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_Str  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThreadState_DeleteCurrent   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyGILState_GetThisThreadState (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_GetAttrString        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyMem_Free    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_Restore (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyType_IsSubtype      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyModule_AddObject    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCapsule_SetPointer  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyTuple_New   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_SetAttr      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_IsInstance   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyException_SetTraceback      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyNumber_Float        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyUnicode_FromFormat  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyList_Append (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_Contains       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_Next   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyList_Size   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyTuple_Size  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyNumber_Long (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyBuffer_Release      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_Format  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_CallObject   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyFloat_FromDouble    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyFloat_AsDouble      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyUnicode_DecodeUTF8  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _Py_Dealloc   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyByteArray_AsString  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyList_New    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyImport_ImportModule (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyNumber_Check        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _PyObject_GetDictPtr  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyUnicode_FromString  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyIndex_Check (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: Py_GetVersion (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCapsule_SetContext  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyFrame_GetLineNumber (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThread_tss_get      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyBytes_Size  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PySequence_Check      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyList_GetItem        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyException_SetContext        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_Clear   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_HasAttrString        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyWeakref_NewRef      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_New    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_SetString       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCapsule_GetContext  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThreadState_Get     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_SetItem      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCapsule_New (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyMem_Calloc  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_SetAttrString        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyGILState_Release    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCapsule_GetPointer  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyNumber_Xor  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThread_tss_alloc    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyBytes_AsString      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_GetItemWithError       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThread_tss_set      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_GetItem      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyType_Ready  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyEval_SaveThread     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PySequence_GetItem    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyNumber_Invert       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_ClearWeakRefs        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_GC_UnTrack   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PySequence_Size       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyLong_FromLong       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyEval_GetBuiltins    (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_WriteUnraisable (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_RichCompareBool      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyNumber_Or   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyModule_Create2      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThread_tss_create   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyLong_AsUnsignedLong (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyLong_FromSize_t     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyFrame_GetBack       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyUnicode_AsEncodedString     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_Occurred        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_Copy   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_Fetch   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThreadState_New     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _PyThreadState_UncheckedGet   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: _PyType_Lookup        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_CallFunctionObjArgs  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_Size   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PySequence_Tuple      (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyThreadState_Clear   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyDict_DelItemString  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyUnicode_AsUTF8AndSize       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyGILState_Ensure     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyObject_Malloc       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCMethod_New (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyErr_Print   (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyCapsule_GetName     (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyTuple_GetItem       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyFrame_GetCode       (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyException_SetCause  (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyUnicode_AsUTF8String        (./gsm_python.cpython-311-x86_64-linux-gnu.so)
undefined symbol: PyBytes_AsStringAndSize       (./gsm_python.cpython-311-x86_64-linux-gnu.so)

```

Esta clase lib/demapping/universal_ctrl_chans_demapper_impl.cc por lo visto tampoco es encontrada al ejecutar su envoltorio desde python..
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/dist-packages/gnuradio/gsm/gsm_sdcch8_demapper.py", line 89, in __init__
    self.gsm_universal_ctrl_chans_demapper_0 = gsm.universal_ctrl_chans_demapper(
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'gnuradio.gsm' has no attribute 'universal_ctrl_chans_demapper'
```

ptrkrysik commented:
Thank you for the effort - I'm currently using this pull request as a reference for porting that is ongoing in the porting_to_3.11 branch. The plan is to selectively port blocks discarding some unused parts of the project and maybe
find some new way of testing the project, as travis-ci is not as easily available for free software projects as it used to be.

I'm closing the pull request here (I admit it took really long time).

alphafox02 commented:
I’ve been using this PR for some time without issue on 3.10 and it yields results, I do change the port on livemon back to 4729 though. I believe the decode feature/script is not working though or so I’ve been told. Scanner and livemon work great though. I have it running on 24.04 with Python 3.12.

alphafox02 commented:
I’ll give your 3.11 branch a try with 3.10.11 GR. There’s still GSM present here for testing.

ptrkrysik commented:
I've added a branch from @bkerler 's repository (bkerler_fork). To make grgsm_decode script working again I need to port some filtering blocks as they are used in it. I didn't check @bkerler 's version but in my porting_to_3.11 branch the script doesn't work yet for sure.

alphafox02 commented:
Excited to see the repo being updated. What a great project. Thank you and @ and others for the  awesome work. That’s great to hear decode is being looked at.
